tests/publiccloud/azure_aitl.pm

```diff
@@ -0,0 +1,129 @@
+use JSON;
+use XML::LibXML;
+
+sub run {
+    my ($self, $args) = @_;
+    select_serial_terminal;
+    my $job_id = get_current_job_id();
+
+    assert_script_run('az version');
+
+    my $provider = $self->provider_factory();
+
+    my $region = get_var('PUBLIC_CLOUD_REGION', 'westus1');
+    my $resource_group = "openqa-aitl-$job_id";
+    my $subscription_id = $provider->provider_client->subscription;
+
+    my $aitl_client_version = "20241118.1";
+    my $aitl_image_gallery = "test_image_gallery";
+    my $aitl_image_version = "latest";
+    my $aitl_job_name = "openqa-aitl-$job_id";
+    my $aitl_manifest = "shared_gallery.json";
+    my $aitl_image_name = $provider->generate_azure_image_definition();
+
+    my $aitl_get_options = "-s $subscription_id -r $resource_group -n $aitl_job_name";
+
+    my $openqa_url = get_var('OPENQA_URL', get_var('OPENQA_HOSTNAME'));
+    my $created_by = "$openqa_url/t$job_id";
+    my $tags = "openqa-aitl=$job_id openqa_created_by=$created_by openqa_var_server=$openqa_url";
+
+    # Get the AITL script
+    assert_script_run("curl https://raw.githubusercontent.com/microsoft/lisa/refs/tags/$aitl_client_version/microsoft/utils/aitl/aitl.py -o /tmp/aitl.py");
+
+    # Configure default location and create Resource group
+    assert_script_run("az configure --defaults location=$region");
+    assert_script_run("az group create -n $resource_group --tags '$tags'");
+
+    # Get manifest from data folder
+    assert_script_run("curl " . data_url("publiccloud/aitl/$aitl_manifest") . " -o /tmp/$aitl_manifest");
+    assert_script_run("sed -i -e 's/<IMAGE_NAME>/$aitl_image_name/g' -e 's/<IMAGE_VERSION>/$aitl_image_version/g' -e 's/<IMAGE_GALLERY_NAME>/$aitl_image_gallery/g' /tmp/$aitl_manifest");
+
+    # Create AITL Job based on a manifest
+    assert_script_run("python3.11 /tmp/aitl.py job create $aitl_get_options -b @/tmp/$aitl_manifest");
+
+    # Wait a few seconds to give Azure time to create the jobs
+    sleep(10);
+
+    my $status = script_output(qq(python3.11 /tmp/aitl.py job get $aitl_get_options -q "properties.results[].status|{RUNNING:length([?@=='RUNNING']),QUEUED:length([?@=='QUEUED'])}"));
+
+    # Remove the first two non-JSON lines from the status JSON.
+    $status =~ s/^(?:.*\n){1,3}//;
+    my $status_data = decode_json($status);
+
+    # Get AITL job status
+    # AITL Jobs run in parallel so it's possible to have Jobs in all kind of states.
+    # The goal of the loop is to check there are no Jobs Queued or currently Running.
+    while ($status_data->{RUNNING} > 0 || $status_data->{QUEUED} > 0) {
+        sleep(30);
+        $status = script_output(qq(python3.11 /tmp/aitl.py job get $aitl_get_options -q "properties.results[].status|{RUNNING:length([?@=='RUNNING']),QUEUED:length([?@=='QUEUED'])}"));
+        $status =~ s/^(?:.*\n){1,3}//;
+        $status_data = decode_json($status);
+        print("Unfinished AITL Jobs! Running:", $status_data->{RUNNING}, " QUEUED: ", $status_data->{QUEUED});
```

Comment:
maybe we need to replace this with `record_info` ?

Reply:
Not needed. This helps when you check the Live Log on the job, and you can see how many Jobs are scheduled and how much you have to wait for. If it was Record Info, it would appear in the end, at which point you would already know this.